tests/test_rates_at_checkout.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+import shippo
+from shippo.models.components import ServiceGroupCreateRequest, Carriers, \
+    ServiceGroupAccountAndServiceLevel, ServiceGroupType, LiveRateCreateRequest, AddressCompleteCreateRequest, \
+    LineItem, WeightUnit, Parcel, DistanceUnitTemplate
+from tests.helpers_custom import get_carrier_account
+
+
+class TestRatesAtCheckout:
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, api: shippo.Shippo):
+        self._delete_all_service_groups(api)
+
+    def _delete_all_service_groups(self, api: shippo.Shippo):
+        response = api.service_groups.list_service_groups()
+        for sg in response.service_group_list_response:
+            api.service_groups.delete_service_group(sg.object_id)
+
+    def test_create_service_group(self, api: shippo.Shippo):
+        carrier_account = get_carrier_account(api, Carriers.USPS)
+        ups_account_id = carrier_account.object_id
+
+        # TODO: update spec so that ServiceLevel* enum models are generated
```

Comment:
is this linked to a ticket in backlog?

Reply:
it'll be the same ticket as this, just waiting for a spec PR to be merged